framework/src/headers/ip/v6/icmp/ndp_options.rs

```diff
@@ -0,0 +1,212 @@
+use std::fmt;
+use std::slice;
+
+use num::FromPrimitive;
+
+use headers::ip::v6::icmp::ndp::NdpMessageContents;
+use headers::mac::MacAddress;
+
+#[derive(FromPrimitive, Debug, PartialEq, Hash, Eq, Clone, Copy)]
+#[repr(u8)]
+pub enum NdpOptionType {
+    SourceLinkLayerAddress = 1,
+    TargetLinkLayerAddress = 2,
+    PrefixInformation = 3,
+    RedirectHeader = 4,
+    MTU = 5,
+}
+
+impl fmt::Display for NdpOptionType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            NdpOptionType::SourceLinkLayerAddress => write!(f, "Source Link Layer Address"),
+            NdpOptionType::TargetLinkLayerAddress => write!(f, "Target Link Layer Address"),
+            NdpOptionType::PrefixInformation => write!(f, "Prefix Information"),
+            NdpOptionType::RedirectHeader => write!(f, "Redirect Header"),
+            NdpOptionType::MTU => write!(f, "MTU"),
+        }
+    }
+}
+
+
+/**
+    MTU NDP Option type
+
+       0                   1                   2                   3
+       0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |     Type      |    Length     |           Reserved            |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |                              MTU                              |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+   Fields:
+
+      Type           5
+
+      Length         1
+
+      Reserved       This field is unused.  It MUST be initialized to
+                     zero by the sender and MUST be ignored by the
+                     receiver.
+
+      MTU            32-bit unsigned integer.  The recommended MTU for
+                     the link.
+
+   Description
+                     The MTU option is used in Router Advertisement
+                     messages to ensure that all nodes on a link use the
+                     same MTU value in those cases where the link MTU is
+                     not well known.
+
+                     This option MUST be silently ignored for other
+                     Neighbor Discovery messages.
+
+                     In configurations in which heterogeneous
+                     technologies are bridged together, the maximum
+                     supported MTU may differ from one segment to
+                     another.  If the bridges do not generate ICMP
+                     Packet Too Big messages, communicating nodes will
+                     be unable to use Path MTU to dynamically determine
+                     the appropriate MTU on a per-neighbor basis.  In
+                     such cases, routers can be configured to use the
+                     MTU option to specify the maximum MTU value that is
+                     supported by all segments.
+*/
+#[derive(Debug)]
+#[repr(C, packed)]
+pub struct MtuOption {
+    option_type: u8,
```

Comment:
@pauljamescleary what are your thoughts on having
```
#[derive(Debug)]
#[repr(C, packed)]
pub struct NDPOptionMeta {
    option_type: u8,
    option_length: u8
}

#[derive(Debug)]
#[repr(C, packed)]
pub struct MtuOption {
    meta: NDPOptionMeta,
    reserved: u16,
    mtu: u32,
}

#[derive(Debug)]
#[repr(C, packed)]
pub struct LinkLayerAddressOption {
   meta: NDPOptionMeta,
    addr: MacAddress,
}
```

That abstracts away the option_type and option_length fields

Reply:
@scrollins good question, I had a brief convo with @drunkirishcoder about the same.  Either way it is repetition, and we just settled on if we had to repeat ourselves it doesn't matter much since it is only 2 fields